dynamic.nix

```diff
@@ -24,7 +24,10 @@ pkgs.mkShell {
             ;;
         esac
         }
-    '';
+    '' ++ (if pkgs.stdenv.targetPlatform.isMacOS then ''
+    # this one is only needed on macOS right now, due to a bug in loading libcrypto.
+    export DYLD_LIBRARY_PATH=$(pkg-config --libs-only-L libcrypto|cut -c 3-)
+    '' else "");
```

Comment:
can this be done with `pkgs.lib.optional`?

can't we just set `DYLD_LIBRARY_PATH = "${openssl.lib}/"` in nix though?